configs/quota/quota.proto

```diff
@@ -0,0 +1,53 @@
+syntax = "proto3";
+
+package quota;
+
+package istio.mixer.config;
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/descriptor.proto";
+
+
+// Modeled on QuotaDescriptor
+// All fields except 
+message Template{
+  // name is a required field for all templates.
+  string name = 1;
+
+  // The set of dimensions that are necessary to describe a specific value cell
+  // for a quota of this type.
+  map<string, ValueType> dimensions = 2; [(istio.mixer.config.request_arg) = true];
```

Comment:
We will have 3 kinds of field
- available only req path (example ListTemplate.CheckExpression)
- available only when defining Types and not during request path (example MetricTemplate.Kind)
- available both when defining Types and also during request path (example MetricTemplate.value)

If we think these 3 combinations are valid, then we would need another annotation to cover the three cases. may be something like (istio.mixer.config.type_arg).

geeknoid@ and I talked about it, I think you also suggested on my PR, to use explicit messages. I have an example for it, https://github.com/guptasu/mixer/pull/6/files#diff-a0ad06bac1ec4482efcd556f66391bd6R21 Thoughts ?